### PR TITLE
Convert `BTConfiguration`, `BTHTTPErrors` to Swift

### PR DESCRIPTION
**Pull request by mattwylder**
### Summary of changes

- Converted `BTConfiguration` and `BTHTTPErrors` to Swift
- Removed `BTErrors`

### Checklist

~- [ ] Added a changelog entry~

- @jaxdesmarais @mattwylder

### CODE REVIEW

File: Sources/BraintreeCoreSwift/BTConfiguration+GraphQL.swift [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Foundation
+
+public extension BTConfiguration {
+    var isGraphQLEnabled: Bool {
+        json?["graphQL"]["url"].asString()?.count ?? 0 > 0
```
Comment (sshropshire) [inline Sources/BraintreeCoreSwift/BTConfiguration+GraphQL.swift:5]:
```suggestion
        (json?["graphQL"]["url"].asString()?.count ?? 0) > 0
```
nit 🧶

File: Sources/BraintreeCoreSwift/BTConfiguration.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+/// Contains information specific to a merchant's Braintree integration
+@objcMembers public class BTConfiguration: NSObject {
```
Comment (sshropshire) [inline Sources/BraintreeCoreSwift/BTConfiguration.swift:4]: random q: Do the classes that aren't meant for the merchant to use directly need to have ObjC annotations?

**Review by sshropshire — APPROVED** [review]
🌊

File: Sources/BraintreeCoreSwift/BTConfiguration.swift (hunk 2 above)
Comment (mattwylder) [reply Sources/BraintreeCoreSwift/BTConfiguration.swift:4]: For the moment, yes. That way we can use the code in our own un-converted Objc code. But once all our stuff is converted we could remove it in certain classes.

**Review by kaylagalway — APPROVED** [review]
:+1::+1:

File: Sources/BraintreePayPal/BTPayPalClient.m [hunk 3]
```diff
@@ -143,7 +143,7 @@ - (void)tokenizePayPalAccountWithPayPalRequest:(BTPayPalRequest *)request comple
                 if (error.code == NETWORK_CONNECTION_LOST_CODE) {
                     [self.apiClient sendAnalyticsEvent:@"ios.paypal.tokenize.network-connection.failure"];
                 }
-                NSString *errorDetailsIssue = ((BTJSON *)error.userInfo[BTHTTPJSONResponseBodyKey][@"paymentResource"][@"errorDetails"][0][@"issue"]).asString;
+                NSString *errorDetailsIssue = ((BTJSON *)error.userInfo[BTHTTPError.jsonResponseBodyKey][@"paymentResource"][@"errorDetails"][0][@"issue"]).asString;
```
[the remark this answers is not in the document]
Comment (jaxdesmarais) [reply Sources/BraintreePayPal/BTPayPalClient.m:146]: Yeah, eventually all of the modules will be converted to Swift and we should be able to clean these instances up a bit more.

**Pull request merged**